statistics/statistics.go

```diff
@@ -302,34 +467,95 @@ func (p *Plugin) Process(contentType string, content []byte, config map[string]c
 		log.Printf("Error decoding: error=%v content=%v", err, content)
 		return "", nil, err
 	}
+	var results []plugin.MetricType
+	times := make([]time.Time, p.bufferMaxSize)
 
+	metricNamespace := make(map[string][]plugin.MetricType)
 	for _, metric := range metrics {
-		log.Printf("Data received %v", metric.Data())
-		p.insertInToBuffer(metric.Data(), metric.Namespace().Strings())
+		ns := concatNameSpace(metric.Namespace().Strings())
+		if plugins, ok := metricNamespace[ns]; ok {
+			plugins = append(plugins, metric)
+			metricNamespace[ns] = plugins
+		} else {
+			metricNamespace[ns] = []plugin.MetricType{metric}
+		}
 	}
 
-	p.updateCounters()
+	for k, v := range metricNamespace {
+		var startTime time.Time
+		var stopTime time.Time
+		unit := v[0].Unit()
+
+		sort.Sort(byTimestamp(v))
+		for _, metric := range v {
+			switch reflect.ValueOf(metric.Data()).Kind() {
+			default:
+				st := fmt.Sprintf("Unkown data received: Type %T", reflect.ValueOf(metric.Data()).Kind())
+				return "", nil, errors.New(st)
+			case reflect.Float64:
```

Comment:
each switch is pretty same except value. Do you like to take the similar part to the bottom?